src/XRP/model/xrp-path-element.ts

```diff
@@ -19,8 +20,28 @@ export default class XrpPathElement {
   public static from(pathElement: PathElement): XrpPathElement {
     const account = pathElement.getAccount()?.getAddress()
     const currency = pathElement.getCurrency()
+    if (account && currency) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'PathElement protobuf should not contain both `account` and `currency` fields.',
```

Comment:
like in https://github.com/xpring-eng/Xpring-JS/pull/716, i wonder if it's worth having a general error message for this...i think this is cool for now though. 

since this is a common pattern it may be nice to have a more declarative function to use for this check (perhaps an interface like `assertOneOfFields(protoBuf, keys: [])` that throws an error if the invariant is violated)

Reply:
so would `assertOneOfFields` basically take in an array of required fields and throw if any one of them is missing?  I think this is a good idea, in conjunction with a method on `XrpError` that allows for filling in the missing field only.  Mayukha you can choose when/how/if to do this and perform a refactor at some point in your chain.